File: code-examples/php/indexes/compound.php
<?php

require_once __DIR__ . "/vendor/autoload.php";

// Replace the following with your MongoDB deployment's connection string
$uri = 'mongodb+srv://<user>:<password>@<cluster-url>?w=majority';

// Create a new client and connect to MongoDB
$client = new MongoDB\Client($uri);

try {
    // Connect to the "sample_mflix" database
    $database = $client->sample_mflix;
    
    // Access the database's "movies" collection
    $collection = $database->movies;

    // Create an ascending index on the "type" and "genre" fields in the "movies" collection
    $collection->createIndex(['type' => 1, 'genre' => 1]);

    // Define a query to find movies in the "Drama" genre
    $query = ['type' => 'movie', 'genre' => 'Drama'];

    // Define sorting criteria for the query results
    $options = [
        'sort' => ['type' => 1, 'genre' => 1],
        'projection' => ['_id' => 0, 'type' => 1, 'genre' => 1],
    ];

    // Execute the query using the defined criteria and options
    $cursor = $collection->find($query, $options);

} catch(Exception $e) {
    echo $e->getMessage();
} finally {
    // As of MongoDB driver 1.5.0, a MongoClient object has no close() method. Connections are implicitly cleaned up when the object is destroyed.
}

?>

File: code-examples/scala/indexes/compound.scala
import org.mongodb.scala._

object MongoApp {

  val mongoClient: MongoClient = MongoClient("mongodb+srv://<user>:<password>@<cluster-url>?writeConcern=majority")
  val database: MongoDatabase = mongoClient.getDatabase("sample_mflix")
  val collection: MongoCollection[Document] = database.getCollection("movies")

  def runMongoApp(): Unit = {
    val indexResultObservable = collection.createIndex(
      Document("type" -> 1, "genre" -> 1)
    )

    indexResultObservable.subscribe((indexName: String) =>
      println(s"Index created: $indexName")
    )

    val query = Document("type" -> "movie", "genre" -> "Drama")
    val sort = Document("type" -> 1, "genre" -> 1)
    val projection = Document("_id" -> 0, "type" -> 1, "genre" -> 1)

    val findObservable = collection
      .find(query)
      .sort(sort)
      .projection(projection)

    findObservable.subscribe(
      (doc: Document) => println(doc.toJson()),
      (e: Throwable) => e.printStackTrace(),
      () => println("Operation Completed")
    )
  }

  def close(): Unit = {
    mongoClient.close()
  }

  def main(args: Array[String]): Unit = {
    try {
      runMongoApp()
    } catch {
      case e: Throwable => e.printStackTrace()
    } finally {
      close()
    }
  }

}
